src/runtime/pyobject.cs

```diff
@@ -134,21 +136,43 @@ protected virtual void Dispose(bool disposing)
         {
             if (!disposed)
             {
-                if (Runtime.Py_IsInitialized() > 0 && !Runtime.IsFinalizing)
+                disposed = true;
+
+                if (disposing)
+                {
+                    try
+                    {
+                        if (Runtime.Py_IsInitialized() > 0 && !Runtime.IsFinalizing)
+                        {
+                            IntPtr gs = PythonEngine.AcquireLock();
+                            try
+                            {
+                                Runtime.XDecref(obj);
+                                obj = IntPtr.Zero;
+                            }
+                            finally
+                            {
+                                PythonEngine.ReleaseLock(gs);
+                            }
+                        }
+                    }
+                    catch
```

Comment:
isn't is better to explicitly fail, instead of "bypassing" the exception handling with an empty `catch {}`? What issue is this solving that `disposed` and `disposing` flags are not taking care of?